python/ray/data/block.py

```diff
@@ -38,6 +39,33 @@
 MaybeBlockPartition = Union[Block, BlockPartition]
 
 
+@DeveloperAPI
+class BlockExecStats:
+    """Execution stats for this block.
+
+    Attributes:
+        wall_time_s: The wall-clock time it took to compute this block.
+        cpu_time_s: The CPU time it took to compute this block.
+        node_id: A unique id for the node that computed this block.
+    """
+
+    # Placeholder for block ops not yet instrumented.
+    TODO = None
+
+    # TODO(ekl) add a builder to fill these out in a simpler way.
+    def __init__(self):
+        self.wall_time_s: Optional[float] = None
+        self.cpu_time_s: Optional[float] = None
```

Comment:
Any reason for not using the `Timer` context manager for these as well?

Reply:
I was planning to address in a future PR (adding a builder).